src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArraysTests.cs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class AvoidConstArraysTests
+    {
+        #region C# Tests
+
+        [Fact]
+        public async Task CA1849_CSharp_IdentifyConstArrays()
+        {
+            // Implicit initialization check
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+
+public class A
+{
+    public void B()
+    {
+        Console.WriteLine({|CA1849:new[]{ 1, 2, 3 }|});
+    }
+}
+", @"
+using System;
+
+public class A
+{
+    private static readonly int[] value = new[]{ 1, 2, 3 };
+
+    public void B()
+    {
+        Console.WriteLine(value);
+    }
+}
+");
+
+            // Explicit initialization check
+            await VerifyCS.VerifyCodeFixAsync(@"
```

Comment:
[suggestion] Could we group the test by "function" instead of having these group comments? Also, merging C# and VB.NET in the same functional test.

Reply:
Sure. I'm on it.